engine/src/main/java/org/terasology/identity/storageServiceClient/ServiceAPIRequest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.identity.storageServiceClient;
+
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.math.BigInteger;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * Utility class to perform requests to the service API.
+ */
+final class ServiceAPIRequest {
+
+    private static final Gson GSON = new GsonBuilder().registerTypeHierarchyAdapter(BigInteger.class, BigIntegerBase64Serializer.getInstance()).create();
+
+    private ServiceAPIRequest() {
+    }
+
+    private static boolean isSuccessful(int code) {
+        return code >= 200 && code < 300;
+    }
+
+    private static void parseError(HttpURLConnection conn) throws IOException, StorageServiceException {
+        try (InputStream errResponse = conn.getErrorStream()) {
+            try {
+                throw new StorageServiceException(GSON.fromJson(new InputStreamReader(errResponse), ErrorResponseData.class).error);
+            } catch (RuntimeException e) {
```

Comment:
Can we be more specific here? Who could even throw during throwing an exception?